tests/testthat/testCorrectForMemoryEffect.R

```diff
@@ -270,7 +270,71 @@ test_that("different numbers of injections for the block 1 standards does not ca
   )
   
-  actual <- correctForMemoryEffect(dataset)
+  actual <- correctForMemoryEffect(dataset4)
   
   expect_is(actual, "list")
-})
\ No newline at end of file
+})
+
+# in this dataset:
+# d18O: m.tilde = 0.082085
+# dD: m.tilde = 0.2231302
+dataset5 <- tribble(
+  ~Line, ~`Identifier 1`, ~`Inj Nr`, ~`d(18_16)Mean`, ~block, ~`d(D_H)Mean`,
+  #--- / -------------- / -------- / -------------- / ----- / -------------
+  1,     "WU",            1,         -9.179,           1,    -62.145,
+  2,     "WU",            2,         -9.933,           1,    -76.017,
+  3,     "WU",            3,         -9.994,           1,    -79.111,
+  4,     "WU",            4,         -9.999,           1,    -79.802,
+  5,     "WU",            5,         -10,              1,    -79.956,
+  6,     "WU",            6,         -10,              1,    -79.990,
+  7,     "WU",            7,         -10,              1,    -79.998,
+  8,     "WU",            8,         -10,              1,    -79.999,
+  9,     "WU",            9,         -10,              1,    -80,
+  10,    "WU",            10,        -10,              1,    -80,
+  11,    "Std1",          1,         -28.358,          1,    -204.299,
+  12,    "Std1",          2,         -29.865,          1,    -232.034,
+  13,    "Std1",          3,         -29.989,          1,    -238.223,
+  14,    "Std1",          4,         -29.999,          1,    -239.603,
+  15,    "Std1",          5,         -30,              1,    -239.912,
+  16,    "Std1",          6,         -30,              1,    -239.980,
+  17,    "Std2",          1,         -39.179,          1,    -302.150,
+  18,    "Std2",          2,         -39.933,          1,    -316.017,
+  19,    "Std2",          3,         -39.994,          1,    -319.111,
+  20,    "Std2",          4,         -39.999,          1,    -319.802,
+  21,    "Std2",          5,         -40,              1,    -319.956,
+  22,    "Std2",          6,         -40,              1,    -319.990,
+  23,    "A",             1,         -35.410,          NA,   -288.923,
+  24,    "A",             2,         -35.034,          NA,   -281.992,
+  25,    "A",             3,         -35.003,          NA,   -280.444,
+  26,    "A",             4,         -35.,             NA,   -280.099,
+  27,    "A",             5,         -35.,             NA,   -280.022,
+  28,    "A",             6,         -35.,             NA,   -280.005
+)
+
+test_that("test that no NA mean memory coefficients are kept", {
+
+  actual <- calculateMemoryCoefficients(dataset5)
+
+  expect_equal(nrow(actual), 6)
+  expect_equal(sum(is.na(c(actual$memoryCoeffD18O, actual$memoryCoeffDD))), 0)
+
+})
+
+test_that("test that number of memory coefficients fits sample data", {
+
+  dataset6 <- add_row(dataset5,
+                      Line = 29, `Identifier 1` = "A", `Inj Nr` = 7,
+                      `d(18_16)Mean` = -35., `d(D_H)Mean` = -280.001,
+                      block = NA)
```

Comment:
It is best-practice to explicitely define all values used for testing. In this case it would be better to explicitely define a data frame (or tibble) that satisfies the condition you want to check here. (i.e. add a dataset6)

Reason: Imagine dataset5 is changed. Now this test may no longer work and may need to be changed as well. Also this `add_row` command may contain a bug or a bug might be introduced unknowingly in the future.

Reply:
Thanks for your comment on this, you are absolutely right. I adopted this in 
3b6a306.